### PR TITLE
My Code Review suggestions.

### PR DESCRIPTION
**Pull request by Jasrah85**
**Analysis of the program:**
I believe the TravelTracker has a lot of potential for being a very useful and in-depth app. There are a lot of connections that are not drawn yet, but I can see that the team has some well thought out plans for the future functioning of this app. The UI design is simple and intuitive with a pleasant color scheme. The font for the description is a little small on my emulator; I would consider enlarging it slightly.
**Was the program available in UC Github on time?**
Yes.
**Is the program documented/commented well enough for you to understand?**
There were not really any comments in the code other than TODOs, but the variable and class names are well chosen and give a good indication of what they are for.
**Does the program compile?**
Yes. The code compiles but was having a problem running on the emulator. I resolved this adding compileOptions to the build.gradle.

Three specific technical concepts that I learned from this code review:

- How to do a Code Cleanup in Android Studio

In the menu: Analyze > Code Cleanup

- How to display a Toast message on the screen

Toast.makeText(activity, "Toast text!", Toast.LENGTH_LONG).show()

- How to set an onClickListener to a button.
I initially had this placed outside of the onActivityCreated function, then realized that is where it would need to be initialized.

### CODE REVIEW

File: app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainViewModel.kt [hunk 1]
```diff
@@ -9,15 +9,15 @@ import uc.edu.ozturkkl.traveltracker.services.LocationService
...
+    var locationService = LocationService()
     private lateinit var firestore : FirebaseFirestore
 
     init {
         loadPredictions(address)
     }
 
-    fun loadPredictions(address : String) {
-        _locations = locationService.loadPredictions(address);
+    private fun loadPredictions(address : String) {
```
Comment (Jasrah85) [inline app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainViewModel.kt:19]: Made loadPredictions a private function.

File: app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainViewModel.kt [hunk 2]
```diff
@@ -9,15 +9,15 @@ import uc.edu.ozturkkl.traveltracker.services.LocationService
 class MainViewModel : ViewModel() {
     private var _locations: MutableLiveData<ArrayList<PredictionResponse>> = MutableLiveData()
     var address: String = String()
-    var locationService = LocationService();
+    var locationService = LocationService()
```
Comment (Jasrah85) [inline app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainViewModel.kt:12]: Removed semicolon.

File: app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainFragment.kt [hunk 3]
```diff
@@ -26,7 +28,14 @@ class MainFragment : Fragment() {
...
         super.onActivityCreated(savedInstanceState)
         viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)
-        // TODO: Use the ViewModel
+        floatingActionButton3.setOnClickListener{
+            addLocation()
+        }
     }
 
+    private fun addLocation() {
+            Toast.makeText(activity, "Replace this action with the Location Addition Event!", Toast.LENGTH_LONG).show()
```
Comment (Jasrah85) [inline app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainFragment.kt:37]: Screen output placeholder that sends a Toast message to the homescreen, showing the button was clicked.

File: app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainFragment.kt [hunk 4]
```diff
@@ -26,7 +28,14 @@ class MainFragment : Fragment() {
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
         viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)
-        // TODO: Use the ViewModel
+        floatingActionButton3.setOnClickListener{
+            addLocation()
+        }
```
Comment (Jasrah85) [inline app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainFragment.kt:33]: Added an OnClickListener to the main screen button with a placeholder function.

File: app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainFragment.kt [hunk 5]
```diff
@@ -6,7 +6,9 @@ import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Toast
 import uc.edu.ozturkkl.traveltracker.R
+import kotlinx.android.synthetic.main.main_fragment.*
```
Comment (Jasrah85) [inline app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainFragment.kt:11]: Synthetic import for the screen layout to use the button.

File: app/src/main/java/uc/edu/ozturkkl/traveltracker/dto/UserDTO.kt [hunk 6]
```diff
@@ -1,6 +1,3 @@
 package uc.edu.ozturkkl.traveltracker.dto
 
-data class User(var id : Int, var firstName : String, var lastName : String, var username : String, var password : String) {
-
-
-}
\ No newline at end of file
+data class User(var id : Int, var firstName : String, var lastName : String, var username : String, var password : String)
```
Comment (Jasrah85) [inline app/src/main/java/uc/edu/ozturkkl/traveltracker/dto/UserDTO.kt:3]: Android Studio Code Cleanup.

File: app/src/main/java/uc/edu/ozturkkl/traveltracker/RetrofitClientInstance.kt [hunk 7]
```diff
@@ -5,13 +5,13 @@ import retrofit2.converter.gson.GsonConverterFactory
...
-    private var retrofit: Retrofit? = null;
-    private const val BASE_URL = "https://maps.googleapis.com/maps/";
+    private var retrofit: Retrofit? = null
+    private const val BASE_URL = "https://maps.googleapis.com/maps/"
 
     val retrofitInstance : Retrofit?
         get() {
             if (retrofit == null) {
-                retrofit = retrofit2.Retrofit.Builder()
+                retrofit = Retrofit.Builder()
```
Comment (Jasrah85) [inline app/src/main/java/uc/edu/ozturkkl/traveltracker/RetrofitClientInstance.kt:14]: Removed redundant qualifier name.

File: app/src/main/java/uc/edu/ozturkkl/traveltracker/RetrofitClientInstance.kt [hunk 8]
```diff
@@ -5,13 +5,13 @@ import retrofit2.converter.gson.GsonConverterFactory
 
 object RetrofitClientInstance {
 
-    private var retrofit: Retrofit? = null;
-    private const val BASE_URL = "https://maps.googleapis.com/maps/";
+    private var retrofit: Retrofit? = null
+    private const val BASE_URL = "https://maps.googleapis.com/maps/"
```
Comment (Jasrah85) [inline app/src/main/java/uc/edu/ozturkkl/traveltracker/RetrofitClientInstance.kt:9]: Removed unnecessary semicolons.

File: app/build.gradle [hunk 9]
```diff
@@ -23,25 +23,29 @@ android {
...
+    androidTestImplementation 'androidx.test.ext:junit:1.1.1'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
+    implementation 'com.google.android.material:material:1.1.0'
+    implementation 'com.squareup.retrofit2:retrofit:2.7.1'
+    implementation 'com.squareup.retrofit2:converter-gson:2.7.1'
     implementation 'com.google.android.gms:play-services-location:17.0.0'
-    implementation 'com.google.firebase:firebase-analytics:17.2.2'
-    implementation 'com.google.firebase:firebase-firestore:21.4.1'
+    implementation 'com.google.firebase:firebase-analytics:17.3.0'
+    implementation 'com.google.firebase:firebase-firestore:21.4.2'
```
Comment (Jasrah85) [inline app/build.gradle:48]: Updated some dependencies.

File: app/build.gradle [hunk 10]
```diff
@@ -23,25 +23,29 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
```
Comment (Jasrah85) [inline app/build.gradle:29]: Added compileOptions to allow the app to run on the emulator.

File: app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainViewModel.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainViewModel.kt:19]: good

File: app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainViewModel.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainViewModel.kt:12]: Good... an old habit to break.  :)

File: app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainFragment.kt (hunk 5 above)
Comment (discospiff) [reply app/src/main/java/uc/edu/ozturkkl/traveltracker/ui/main/MainFragment.kt:11]: Synthetic imports: very good; save a lot of typing.

**Comment by discospiff** [comment]
Several good suggestions here that clean up the codebase and make the program more readable.  I recommend merging these changes.

**Pull request closed without merging**